docs/stability-tiers.md

```diff
@@ -0,0 +1,194 @@
+# Tiers of Support in Wasmtime
+
+Wasmtime's support for platforms and features can be distinguished with three
+different tiers of support. The description of these tiers are intended to be
+inspired by the [Rust compiler's support tiers for
+targets](https://doc.rust-lang.org/rustc/target-tier-policy.html) but are
+additionally tailored for Wasmtime. Wasmtime's tiered support additionally
+applies not only to platforms/targets themselves but additionally features
+implemented within Wasmtime itself.
+
+The purpose of this document is to provide a means by which to evaluate the
+inclusion of new features and support for existing features within Wasmtime.
+This should not be used to "lawyer" a change into Wasmtime on a precise
+technical detail or similar since this document is itself not 100% precise and
+will change over time.
+
+## Current Tier Status
+
+For explanations of what each tier means see below.
+
+#### Tier 1
+
+* Target - `x86_64-apple-darwin`
+* Target - `x86_64-pc-windows-msvc`
+* Target - `x86_64-unknown-linux-gnu`
+* WASI Proposal - `wasi_snapshot_preview1`
+* WASI Proposal - `wasi_unstable`
+* WebAssembly Proposal - `bulk-memory`
+* WebAssembly Proposal - `reference-types`
+* WebAssembly Proposal - `simd`
+
+#### Tier 2
+
+* Target - `aarch64-unknown-linux-gnu`
+* Target - `s390x-unknown-linux-gnu`
+* Target - `x86_64-pc-windows-gnu`
+* WebAssembly Proposal - `memory64`
+* WebAssembly Proposal - `multi-memory`
+
+#### Tier 3
+
+* Target - `aarch64-apple-darwin`
+* WASI Proposal - `wasi-nn`
```

Comment:
I think wasi-nn could arguably be in tier 2.

Reply:
Given the current list of requirements sketched out I believe that `wasi-nn` is still largely missing tests. There's one example that is compiled and run but I think that we'd want more of a unit-test-like-suite otherwise.

At least personally this sort of tracks with my gut feeling about `wasi-nn` and `wasi-crypto` where they're present but don't feel all-that-well integrated/first-class. WASI itself isn't really all that first-class by requiring an extra `wasmtime-wasi` crate which hasn't had much time put into its API design for stability too, though. I don't know how best to think about WASI proposals myself.